sonar-plugin-api/src/main/java/org/sonar/api/ExtensionProvider.java

```diff
@@ -56,7 +56,9 @@
  * 
  *
  * @since 2.3
+ * @deprecated since 6.0 should no more be used
  */
+@Deprecated
 public abstract class ExtensionProvider implements Extension {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)